apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -434,21 +434,37 @@ mod helpers {
 
             // The body of the request might include references to input arguments / sibling fields
             // that will need to be handled, so we extract any referenced variables now
-            let body_parameters = extract_params_from_body(connector)?;
-
-            let parameters = connector
+            let body_parameters = connector
                 .transport
-                .connect_template
-                .parameters()
-                .map_err(|e| {
-                    FederationError::internal(format!(
-                        "could not extract path template parameters: {e}"
-                    ))
-                })?;
+                .body
+                .as_ref()
+                .map(extract_params_from_selection)
+                .transpose()?
+                .unwrap_or_default();
+
+            // The HTTP body might contain references to $this, so we grab those usages as well
+            let path_parameters =
+                connector
+                    .transport
+                    .connect_template
+                    .parameters()
+                    .map_err(|e| {
+                        FederationError::internal(format!(
+                            "could not extract path template parameters: {e}"
+                        ))
+                    })?;
+
+            // Most importantly, the actual selection might make use of the $this variable, so we grab them too
```

Comment:
"most importantly" is interesting commentary :grin: — this will be the least frequently used, at least until we support transport-less connectors like `area: @resolve(selection: "$this->width->mul($this.height)")`

Reply:
Oh fair. I'll edit the comment in a future PR!